go/consensus/cometbft/apps/roothash/state/state.go

```diff
@@ -303,6 +353,56 @@ func (s *MutableState) SetRuntimeState(ctx context.Context, state *roothash.Runt
 	if err := s.ms.Insert(ctx, ioRootKeyFmt.Encode(&state.Runtime.ID), ioRoot); err != nil {
 		return api.UnavailableStateError(err)
 	}
+
+	// Clean previously stored state and I/O roots if we're over the maximum.
+	params, err := s.ConsensusParameters(ctx)
+	var maxStored uint32
+	if err == nil {
+		maxStored = params.MaxPastRootsStored
+	}
+
+	roots, err := s.PastRoots(ctx, state.Runtime.ID)
```

Comment:
It doesn't seem optimal that you load all of the past roots just to (in the most common case) add one root and remove one root. You could probably just blindly remove the root at round `state.CurrentBlock.Header.Round - MaxPastRoots`.

Reply:
Done, please take a look :)